### DISCUSSION TITLE
User Defined Scripts not working

### DISCUSSION THREAD

**Discussion by skyEnrique** [ant-media/Ant-Media-Server#6112] [Q&A] [answered]
I have a bash script (vod-thumb-gen.sh) that generates thumbnail from the vod video,

bash script is located at 
/home/ubuntu/ams-scripts

I also run this command as per the tutorial `chmod +x vod-thumb-gen.sh`

When i manually run the bash script, it works fine.
But it doesnt work when ant media executed it. 
AMS app settings:
  "muxerFinishScript": "/home/ubuntu/ams-scripts/vod-thumb-gen.sh"

This log can be seen on the Console Log
`2024-02-20 05:39:40,959 [vert.x-worker-thread-37] INFO  i.a.AntMediaApplicationAdapter - running muxer finish script: /home/ubuntu/ams-scripts/vod-thumb-gen.sh  /usr/local/antmedia/webapps/LiveApp/streams/enrique/FmiykC8dhsD9ViH01708399903952-2024-02-20_05-39-35.064_1080p.mp4`

As stated on the log, looks like it works fine. But i dont see any result.

Here is my bash script
```
#!/bin/bash

# Check if the correct number of arguments is provided
if [ "$#" -ne 1 ]; then
    echo "Usage: $0 <input_mp4_path>"
    exit 1
fi

# Assign the input file path from the command line argument
input_file="$1"

# Extract the file name without extension
filename=$(basename -- "$input_file")
filename_no_ext="${filename%.*}"

# Define the output file path with the same name as the input file
output_thumbnail="$filename_no_ext.jpg"

# Run the ffmpeg command and print the full path of the output file
ffmpeg -y -i "$input_file" -ss 00:00:05 -vframes 1 -vf "scale=640:-1" "$output_thumbnail"

# Print the full path of the output file
echo "Thumbnail generated: $(pwd)/$output_thumbnail
```

**Comment by muratugureminoglu** [comment, marked as answer]
Hi @skyEnrique 

When I try the script you shared on my side, the thumbnail file is generated without any problem with ffmpeg version 4.4.2. 
 If the ffmpeg version you are using is lower than 4.4.2 and you are trying it in a GUI-less environment, this script may not work, and you may get the error "Error: unable to open display". 

My steps:

1. Enable mp4 recording in **Dashboard > Your App > Settings > Record Live Streams as MP4**
2. Edit or add the following line according to your path in **Dashboard > Your App > Settings > Advance settings**
​
  "muxerFinishScript": "/usr/local/antmedia/vod.sh",

3. Send stream over RTMP/WebRTC
4. Stop the stream

That's it. thumbnail file will be generated where you script located.

Could you please check the ffmpeg version on your side?
